### 6.RoBERTa Tokenizer

In [2]:
# 라이브러리
import torch
from transformers import RobertaConfig, RobertaModel, RobertaTokenizer

In [3]:
# 모델 불러오기
model = RobertaModel.from_pretrained('roberta-base')

config.json:   0%|          | 0.00/481 [00:00<?, ?B/s]

model.safetensors:   0%|          | 0.00/499M [00:00<?, ?B/s]

Some weights of RobertaModel were not initialized from the model checkpoint at roberta-base and are newly initialized: ['roberta.pooler.dense.bias', 'roberta.pooler.dense.weight']
You should probably TRAIN this model on a down-stream task to be able to use it for predictions and inference.


In [4]:
# 토크나이저 불러오기
tokenizer = RobertaTokenizer.from_pretrained('roberta-base')

tokenizer_config.json:   0%|          | 0.00/25.0 [00:00<?, ?B/s]

vocab.json:   0%|          | 0.00/899k [00:00<?, ?B/s]

merges.txt:   0%|          | 0.00/456k [00:00<?, ?B/s]

tokenizer.json:   0%|          | 0.00/1.36M [00:00<?, ?B/s]

- 모델과 토크나이저를 불러왔다.

In [5]:
model.config

RobertaConfig {
  "_name_or_path": "roberta-base",
  "architectures": [
    "RobertaForMaskedLM"
  ],
  "attention_probs_dropout_prob": 0.1,
  "bos_token_id": 0,
  "classifier_dropout": null,
  "eos_token_id": 2,
  "hidden_act": "gelu",
  "hidden_dropout_prob": 0.1,
  "hidden_size": 768,
  "initializer_range": 0.02,
  "intermediate_size": 3072,
  "layer_norm_eps": 1e-05,
  "max_position_embeddings": 514,
  "model_type": "roberta",
  "num_attention_heads": 12,
  "num_hidden_layers": 12,
  "pad_token_id": 1,
  "position_embedding_type": "absolute",
  "transformers_version": "4.38.1",
  "type_vocab_size": 1,
  "use_cache": true,
  "vocab_size": 50265
}

- num_attention_heads 12개, 은닉층 12개, 은닉층 크기 768개, 어휘 사전 크기 50265임을 알 수 있다.

In [6]:
# 토크나이저에 입력값 입력
tokenizer.tokenize('It was a great day')

['It', 'Ġwas', 'Ġa', 'Ġgreat', 'Ġday']

- Ġ라는 알 수 없는 문자가 붙어 있는데, 이는 공백을 의미한다.

In [8]:
# 사전에 없는 단어를 이용해서 토크나이저에 입력값 입력
tokenizer.tokenize('Pre-training is important')

['Pre', '-', 'training', 'Ġis', 'Ġimportant']

- Pre-training이라는 단어를 Pre, -, training으로 분리해줬다.